File: backend/app/schemas/estadisticas.py
"""
Schemas para estadísticas y dashboard
"""
from pydantic import BaseModel
from typing import List, Dict, Optional
from datetime import date


class EstadisticasGeneral(BaseModel):
    """Estadísticas generales del sistema"""
    total_inspecciones: int
    pendientes: int
    aprobadas: int
    rechazadas: int
    total_usuarios: int
    total_plantas: int
    total_navieras: int


class InspeccionPorEstado(BaseModel):
    """Conteo de inspecciones por estado"""
    estado: str
    cantidad: int
    porcentaje: float


class InspeccionPorFecha(BaseModel):
    """Inspecciones agrupadas por fecha"""
    fecha: date
    cantidad: int


class InspeccionPorPlanta(BaseModel):
    """Inspecciones por planta"""
    planta: str
    cantidad: int


class InspeccionPorInspector(BaseModel):
    """Inspecciones por inspector"""
    inspector: str
    total: int
    pendientes: int
    aprobadas: int
    rechazadas: int


class DashboardData(BaseModel):
    """Datos completos para el dashboard"""
    estadisticas_generales: EstadisticasGeneral
    por_estado: List[InspeccionPorEstado]
    por_fecha: List[InspeccionPorFecha]  # Últimos 30 días
    por_planta: List[InspeccionPorPlanta]
    por_inspector: List[InspeccionPorInspector]
    fecha_desde: date
    fecha_hasta: date

File: frontend/src/api/estadisticas.ts
import axios from './axios';

interface EstadisticasGeneral {
  total_inspecciones: number;
  total_aprobadas: number;
  total_rechazadas: number;
  total_pendientes: number;
  promedio_tiempo_respuesta?: number;
}

interface InspeccionPorEstado {
  estado: string;
  total: number;
  porcentaje: number;
}

interface InspeccionPorFecha {
  fecha: string;
  total: number;
}

interface InspeccionPorPlanta {
  nombre_planta: string;
  total: number;
}

interface InspeccionPorInspector {
  nombre_completo: string;
  total_inspeccionadas: number;
  aprobadas: number;
  rechazadas: number;
  pendientes: number;
}

export interface DashboardData {
  fecha_desde: string;
  fecha_hasta: string;
  estadisticas_generales: EstadisticasGeneral;
  por_estado: InspeccionPorEstado[];
  por_fecha: InspeccionPorFecha[];
  por_planta: InspeccionPorPlanta[];
  por_inspector: InspeccionPorInspector[];
}

export const estadisticasApi = {
  async getDashboard(params?: {
    fecha_desde?: string;
    fecha_hasta?: string;
  }): Promise<DashboardData> {
    const response = await axios.get('/estadisticas/dashboard', { params });
    return response.data;
  }
};
